### PR TITLE
Implement nanargmax function in keras.ops

### PR DESCRIPTION
**Pull request by shashaka**
Adds keras.ops.nanargmax, which computes the indices of the maximum values along a specified axis while ignoring NaN values in the input tensor.

The operation follows JAX semantics for axis and keepdims, returning integer indices corresponding to the maximum non-NaN values. For slices where all elements are NaN, it returns -1 per reduced slice.

Supported across NumPy, TensorFlow, PyTorch, and JAX backends. Not supported on OpenVINO.

### CODE REVIEW

File: keras/src/backend/numpy/numpy.py [hunk 1]
```diff
@@ -986,6 +986,21 @@ def moveaxis(x, source, destination):
...
+        return argmax(x, axis=axis, keepdims=keepdims)
+
+    nan_mask = np.isnan(x)
+
+    return np.where(
+        np.all(nan_mask, axis=axis, keepdims=keepdims),
+        -1,
+        np.nanargmax(
+            np.where(nan_mask, -np.inf, x), axis=axis, keepdims=keepdims
+        ).astype("int32"),
```
[the remark this answers is not in the document]
Comment (shashaka) [reply keras/src/backend/numpy/numpy.py:1000]: np.where evaluates both sides, so np.nanargmax(x, ...) would still run on all-NaN slices and raise a ValueError.
Replacing NaNs with -np.inf prevents that error, so the inner np.where is necessary.

File: keras/src/ops/numpy.py [hunk 2]
```diff
@@ -5319,6 +5319,65 @@ def moveaxis(x, source, destination):
...
+    Args:
+        x: Input tensor.
+        axis: By default, the index is into the flattened tensor, otherwise
+            along the specified axis.
+        keepdims: If this is set to `True`, the axes which are reduced are left
+            in the result as dimensions with size one. Defaults to `False`.
+
+    Returns:
+        Tensor of indices. It has the same shape as `x`, with the dimension
+        along `axis` removed. NaN values are ignored when computing the maximum.
```
[the remark this answers is not in the document]
Comment (shashaka) [reply keras/src/ops/numpy.py:5354]: I’m keeping the current wording to stay consistent with the existing nanargmin documentation. Since both functions follow the same behavior, I prefer to update them together rather than document this detail in only one place.

File: keras/src/ops/numpy_test.py [hunk 3]
```diff
@@ -5981,6 +5998,44 @@ def test_slogdet(self):
...
+            knp.nanargmax(x_all_nan, axis=0), np.array([-1, -1])
+        )
+        self.assertAllClose(
+            knp.nanargmax(x_all_nan, axis=1), np.array([-1, -1])
+        )
+        self.assertAllClose(
+            knp.nanargmax(x_all_nan, axis=1, keepdims=True),
+            np.array([[-1], [-1]]),
+        )
+
```
[the remark this answers is not in the document]
Comment (shashaka) [reply keras/src/ops/numpy_test.py:6038]: The mixed case is already covered by existing tests, so I’ll keep the current tests as they are.
